Mathlib/Analysis/NormedSpace/Banach.lean

```diff
@@ -353,6 +353,30 @@ theorem coeFn_toContinuousLinearEquivOfContinuous_symm (e : E ≃ₗ[𝕜] F) (h
 
 end LinearEquiv
 
+namespace ContinuousLinearMap
+
+variable [CompleteSpace E]
+
+/-- An injective continuous linear map with a closed range defines a continuous linear equivalence
+between its domain and its range. -/
+noncomputable def rangeEquiv (f : E →L[𝕜] F) (hinj : Injective f) (hclo : IsClosed (range f)) :
+    E ≃L[𝕜] LinearMap.range f :=
+  have : CompleteSpace (LinearMap.range f) := hclo.completeSpace_coe
+  LinearEquiv.toContinuousLinearEquivOfContinuous (LinearEquiv.ofInjective f.toLinearMap hinj) <|
+    (f.continuous.codRestrict fun x ↦ LinearMap.mem_range_self f x).congr fun _ ↦ rfl
+
+@[simp]
+theorem rangeEquiv_apply (f : E →L[𝕜] F) (hinj : Injective f) (hclo : IsClosed (range f)) (x : E) :
+    f.rangeEquiv hinj hclo x = f x :=
+  rfl
+
+@[simp]
+theorem rangeEquiv_eq (f : E →L[𝕜] F) (hinj : Injective f) (hclo : IsClosed (range f)) :
+    (f.rangeEquiv hinj hclo).toContinuousLinearMap = f.rangeRestrict :=
```

Comment:
Please write the LHS as a coercion instead of using `ContinuousLinearEquiv.toContinuousLinearMap`, unless we don't have that coercion, in such case, please ping me.

Reply:
Turns out Lean4's intelligent coercion insertion works here without needing `.toContinuousLinearMap`!